example-conf/mesos-dns/mesos-site.xml

```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<?xml-stylesheet type="text/xsl" href="configuration.xsl"?>
+<!--
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License. See accompanying LICENSE file.
+-->
+
+<!-- Put site-specific property overrides in this file. -->
+
+<configuration>
+  <property>
+    <name>mesos.hdfs.mesosdns</name>
+    <value>true</value>
+    <description>If you want mesos dns to be used at all.</description>
+  </property>
+
+  <property>
+    <name>mesos.hdfs.mesosdns.domain</name>
+    <value>mesos</value>
+    <description>The domain is usually mesos, but if you changed this in Mesos DNS, change it here.</description>
+  </property>
+
+  <property>
+    <name>mesos.hdfs.state.zk</name>
+    <value>master.mesos:2181</value>
+    <description>Zookeeper is used by Hadoop's HA feature. If you colocate zk and mesos masters, you can use the master location here.</description>
```

Comment:
This ZK parameter is used by the HDFS framework scheduler itself, for storing state like the frameworkId and task states.
That's separate from the mesos.hdfs.zkfc.ha.zookeeper.quorum variable that the ZKFC uses to contact ZK for HDFS namenode election, which you incorrectly reference here.
By tweaking these parameters, the framework could use a different ZK cluster (e.g. the Mesos ZK) for its persistent state than the ZK cluster that HDFS uses for its own leader election (e.g. the Hadoop ZK).